gcloud-java-storage/src/main/java/com/google/gcloud/storage/testing/FakeStorageRpc.java

```diff
@@ -91,8 +89,54 @@ public StorageObject create(StorageObject object, InputStream content, Map<Optio
   @Override
   public Tuple<String, Iterable<StorageObject>> list(String bucket, Map<Option, ?> options)
       throws StorageException {
-    potentiallyThrow(options);
-    return null;
+    String preprefix = "";
+    for (Map.Entry<Option, ?> e : options.entrySet()) {
+      if (e.getKey() == StorageRpc.Option.PREFIX) {
+        preprefix = (String)e.getValue();
+        if (preprefix.startsWith("/")) {
+          preprefix = preprefix.substring(1);
+        }
+        continue;
+      } else {
+        throw new UnsupportedOperationException("Unknown option: "+e.getKey());
+      }
+    }
+    final String prefix = preprefix;
+
+    Iterable<StorageObject> iterable = new Iterable<StorageObject>() {
```

Comment:
Any reason for using this anonymous class rather than computing all values to be returned and then returning them? (which seems to be what you are doing in the anonymous class: `init` populates `values` and then `iterator()` returns an iterator over `values`).